File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/create-game/create-game-response.ts
import type { BaseResponse } from '../../../../base/responses/base-response';
import type { CreateGameResponseDto } from './create-game-response-dto';

export class CreateGameResponse implements BaseResponse<CreateGameResponseDto> {
    Data: CreateGameResponseDto;

    constructor(data?: CreateGameResponseDto) {
        this.Data = data || { GameId: '', Name: '', Status: 0 };
    }
}

File: HorseRacing.Contracts/Models/Game/Responses/CreateGame/CreateGameResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Game.Responses.CreateGame
{
    public class CreateGameResponse : BaseResponse<CreateGameResponseDto>
    {
        public CreateGameResponse() : base() { }

        public CreateGameResponse(CreateGameResponseDto data) : base(data) { }
    }
}
